src/utils/date.tsx

```diff
@@ -0,0 +1,48 @@
+import dayjs from 'dayjs';
+import TimeAgo from 'react-timeago'
+import { default as utc } from 'dayjs/plugin/utc';
+import { default as timezone } from 'dayjs/plugin/timezone';
+
+export const formatDateToUserTimezone = (date: string) => {
+  dayjs.extend(utc);
+  dayjs.extend(timezone);
+  // guess user timezone;
+  const userTimezone = dayjs.tz.guess();
+  const formattedDate = dayjs(date).tz(userTimezone).format('YYYY-MM-DD');
+  return formattedDate;
+};
+
+export const formatTimeToUserTimezone = (date: string) => {
+  dayjs.extend(utc);
+  dayjs.extend(timezone);
+  // guess user timezone;
+  const userTimezone = dayjs.tz.guess();
+  const formattedDate = dayjs(date).tz(userTimezone).format('HH:mm');
+  return formattedDate;
+};
+
+export const calculateTimeInGMT = (date: string) => {
+  dayjs.extend(utc);
+  const timeInGMT = `${dayjs(date).utc().format('YYYY-MM-DD HH:mm')} GMT ${dayjs(date).utc().format('Z')}`;
+  return timeInGMT;
+};
+
+export const formatDate = (epoch: number, twoRows = true, now?: number) => {
+  if(!epoch) return <>-<br/>&nbsp;</>;
+  let differenceMs = Date.now() - new Date(epoch).getTime();
+  // @ts-ignore
+  if (differenceMs < 0) return <TimeAgo date={new Date(epoch).getTime()-(2*60*60*1000)} now={now} />
+  if (differenceMs < 24*60*60*1000) {
+  // @ts-ignore
+    return <TimeAgo date={epoch} now={now} />
+  } else {
+    const d = new Date(epoch);
+    const year = d.getFullYear();
+    const month = d.getMonth()+1 < 10 ? `0${d.getMonth()+1}` : d.getMonth()+1;
+    const day = d.getDate() < 10 ? `0${d.getDate()}` : d.getDate();
+    const hours = d.getUTCHours() < 10 ? `0${d.getUTCHours()}` : d.getUTCHours();
+    const minutes = d.getUTCMinutes() < 10 ? `0${d.getUTCMinutes()}` : d.getUTCMinutes();
+    const formatted = <>{`${year}-${month}-${day}`}{twoRows ? <br/> : ' '}{`${hours}:${minutes}`}</>
+    return <>{formatted}</>;
+  }
```

Comment:
I think we can do this solution with less dependencies with dayjs and probably less lines

relative time: https://day.js.org/docs/en/plugin/relative-time (similar to time ago and we already have the dep)
format time: https://day.js.org/docs/en/display/format

The two rows prop i would move to a special component so the util function returns a string. This being a util func maybe we can even separate it out into a couple functions like:

- formatDateToRelativeTime
- formatDateToUTC
- formatTimeToUTC

we actually already have `formatDateToUserTimezone` and formatTimeToUserTimezone` which can return what you are already returning but in the user timezone maybe you want this in UTC.

After having these seperate util funcs then it would be easy to create a component in the components folder that can even be called TimeAgo doing something like the logic in the if statement that you have there.

```
function TimeAgo(props: { twoRows: bool }) {
// if logic implemented here and chooses which util function to use
}
```

Reply:
Created isssue for that: https://github.com/hoprnet/hopr-admin/issues/464